source/solvers/navier_stokes_base.cc

```diff
@@ -567,82 +572,242 @@ NavierStokesBase<dim, VectorType, DofsType>::finish_time_step()
     }
 }
 
+
+template <int dim, typename VectorType, typename DofsType>
+void
+NavierStokesBase<dim, VectorType, DofsType>::multi_stage_preresolution(
+  unsigned int                                      stage,
+  Parameters::SimulationControl::TimeSteppingMethod method)
+{
+  // Very important variable because sum(a_{ij} * k_j) has to be
+  // calculated at each stage
+  auto &previous_k_j_solutions = this->sdirk_vectors.previous_k_j_solutions;
+
+  // sum(a_{ij} *k_j) when we are at least at stage 2
+  // If we are at stage 1, sum_over_previous_stages is set to 0.
+  auto &sum_over_previous_stages = this->sdirk_vectors.sum_over_previous_stages;
+  auto &temp_sum_over_previous_stages =
+    this->sdirk_vectors.temp_sum_over_previous_stages;
+
+  auto &locally_owned_for_calculus =
```

Comment:
Here I would add a comment that first you copy the reference to some of the vectors to enhance readability below. I have nothing agasint that, it is more readable, but I would explain that more clearly :)